metadata/metadata.c

```diff
@@ -327,31 +398,28 @@ PhysicalDisk *create_physical_disk(Metadata *metadata, NVMfTransport *transport,
     physical_disk->size = sector_count * sector_size;
     physical_disk->transport = transport;
 
-    size_t n_ranges = ceil(sector_count / allocator->physical_disk_range_sector_count);
+    size_t n_ranges = ceil(sector_count / metadata->allocator->physical_disk_range_sector_count);
     for (int i = 0; i < n_ranges; i++)
     {
-        _initialize_next_physical_disk_range(metadata, physical_disk, allocator, i);
+        _initialize_next_physical_disk_range(metadata, physical_disk, i);
     }
 
-    if (metadata != NULL)
-    {
-        _add_physical_disk(metadata, physical_disk);
-        _sort_physical_disks_by_unallocated_ranges(metadata);
-    }
+    _add_physical_disk(metadata, physical_disk);
+    _sort_physical_disks_by_unallocated_ranges(metadata);
 
     return physical_disk;
 }
 
-VirtualDisk *_new_virtual_disk(char *name,
-                               VirtualDisk__ErasureCodeProfile ec_profile,
+VirtualDisk *_new_virtual_disk(Metadata *metadata,
+                               char *name,
                                uint64_t size)
 {
     VirtualDisk *virtual_disk = (VirtualDisk *)malloc(sizeof(VirtualDisk));
     virtual_disk__init(virtual_disk);
 
     _generate_and_set_uuid(&virtual_disk->key);
     virtual_disk->name = _memcpy_from_str(name);
-    virtual_disk->ec_profile = ec_profile;
+    virtual_disk->ec_profile = metadata->ec_profile;
```

Comment:
this change doesn't seem necessary right now. You could just continue to pass the EC profile enum and it would be fine.